File: Python/subprogramas/es_matriz_identidad.py

def es_matriz_identidad(matriz):
    for i in range(len(matriz)):
        for j in range(len(matriz[i])):
            if i == j and matriz[i][j] != 1:
                return False
            elif i != j and matriz[i][j] != 0:
                return False
    return True

# Example usage
if __name__ == "__main__":
    matriz = [
        [1, 0, 0, 0],
        [0, 1, 0, 0],
        [0, 0, 1, 0],
        [0, 0, 0, 1]
    ]
    print(f"¿Es la matriz identidad? {es_matriz_identidad(matriz)}")

File: C++/subprogramas/es_matriz_identidad.cpp

#include <iostream>
#include <vector>

using namespace std;

bool es_matriz_identidad(const vector<vector<int>>& matriz) {
    for (size_t i = 0; i < matriz.size(); ++i) {
        for (size_t j = 0; j < matriz[i].size(); ++j) {
            if ((i == j && matriz[i][j] != 1) || (i != j && matriz[i][j] != 0)) {
                return false;
            }
        }
    }
    return true;
}

int main() {
    vector<vector<int>> matriz = {
        {1, 0, 0, 0},
        {0, 1, 0, 0},
        {0, 0, 1, 0},
        {0, 0, 0, 1}
    };
    cout << "¿Es la matriz identidad? " << (es_matriz_identidad(matriz) ? "Sí" : "No") << endl;
    return 0;
}